vendor/bat-native-ads/include/bat/ads/notification_info.h

```diff
@@ -16,14 +16,15 @@ namespace ads {
 
 struct ADS_EXPORT NotificationInfo {
   NotificationInfo();
-  explicit NotificationInfo(const NotificationInfo& info);
+  NotificationInfo(const NotificationInfo& info);
   ~NotificationInfo();
 
   const std::string ToJson() const;
   Result FromJson(
       const std::string& json,
       std::string* error_description = nullptr);
 
+  std::string id;
```

Comment:
Added a new field here but iOS has `BATAdsNotification` which needs the same field:

- [New Readonly Property](https://github.com/brave/brave-core/blob/a0e854a10668d2c55c7644f6d4149c79744102dc/vendor/brave-ios/Ads/BATAdsNotification.h#L19)
- [New Writable Property](https://github.com/brave/brave-core/blob/a0e854a10668d2c55c7644f6d4149c79744102dc/vendor/brave-ios/Ads/BATAdsNotification.mm#L10)
- [Updated initializer](https://github.com/brave/brave-core/blob/a0e854a10668d2c55c7644f6d4149c79744102dc/vendor/brave-ios/Ads/BATAdsNotification.mm#L23)